### ISSUE TITLE
strptime.py Unhandled Exception after computer restart

### ISSUE THREAD

**Issue by movalex** [fail2ban/fail2ban#1529]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): Fail2Ban v0.10.0a2
- OS, including release name/version: OSX 10.11.6
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)
(no customisations for fail2ban.conf)

### The issue:
strptime.py file  raises 'ValueError: year is out of range'  after each computer restart

#### Any additional information
built with `sudo /usr/bin/python setup.py install`

### Configuration, dump and another helpful excerpts
```
fail2ban-client -d 
['set', 'syslogsocket', 'auto']
['set', 'loglevel', 'INFO']
['set', 'logtarget', '/var/log/fail2ban.log']
['set', 'dbfile', '/var/lib/fail2ban/fail2ban.sqlite3']
['set', 'dbpurgeage', '1d']
['add', 'short-ban', 'auto']
['set', 'short-ban', 'usedns', 'warn']
['set', 'short-ban', 'addlogpath', '/var/log/system.log', 'head']
['set', 'short-ban', 'maxretry', 3]
['set', 'short-ban', 'addignoreip', '127.0.0.1']
['set', 'short-ban', 'addignoreip', '192.168.100.0/24']
['set', 'short-ban', 'addignoreip', 'xxx.xxx.xxx.xxx']
['set', 'short-ban', 'logencoding', 'utf-8']
['set', 'short-ban', 'bantime', '600']
['set', 'short-ban', 'ignorecommand', '']
['set', 'short-ban', 'findtime', '7200']
['set', 'short-ban', 'maxlines', '10']
['multi-set', 'short-ban', 'addfailregex', ['^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?(?:error: PAM: )?[aA]uthentication (?:failure|error|failed) for .* from <HOST>( via \\S+)?\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?(?:error: PAM: )?User not known to the underlying authentication module for .* from <HOST>\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?Failed \\S+ for .*? from <HOST>(?: port \\d*)?(?: ssh\\d*)?(: (ruser .*|(\\S+ ID \\S+ \\(serial \\d+\\) CA )?\\S+ (?:[\\da-f]{2}:){15}[\\da-f]{2}(, client user ".*", client host ".*")?))?\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?ROOT LOGIN REFUSED.* FROM <HOST>\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?[iI](?:llegal|nvalid) user .* from <HOST>\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?User .+ from <HOST> not allowed because not listed in AllowUsers\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?User .+ from <HOST> not allowed because listed in DenyUsers\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?User .+ from <HOST> not allowed because not in any group\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?refused connect from \\S+ \\(<HOST>\\)\\s*$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?(?:error: )?Received disconnect from <HOST>: 3: .*: Auth fail(?: \\[preauth\\])?$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?User .+ from <HOST> not allowed because a group is listed in DenyGroups\\s*$', "^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?User .+ from <HOST> not allowed because none of user's groups are listed in AllowGroups\\s*$", '^(?P<__prefix>(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?)User .+ not allowed because account is locked<SKIPLINES>(?P=__prefix)(?:error: )?Received disconnect from <HOST>: 11: .+ \\[preauth\\]$', '^(?P<__prefix>(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?)Disconnecting: Too many authentication failures for .+? \\[preauth\\]<SKIPLINES>(?P=__prefix)(?:error: )?Connection closed by <HOST> \\[preauth\\]$', '^(?P<__prefix>(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?)Connection from <HOST> port \\d+(?: on \\S+ port \\d+)?<SKIPLINES>(?P=__prefix)Disconnecting: Too many authentication failures for .+? \\[preauth\\]$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?(error: )?maximum authentication attempts exceeded for .* from <HOST>(?: port \\d*)?(?: ssh\\d*)? \\[preauth\\]$', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?pam_unix\\(sshd:auth\\):\\s+authentication failure;\\s*logname=\\S*\\s*uid=\\d*\\s*euid=\\d*\\s*tty=\\S*\\s*ruser=\\S*\\s*rhost=<HOST>\\s.*$']]
['set', 'short-ban', 'addjournalmatch', '_SYSTEMD_UNIT=sshd.service', '+', '_COMM=sshd']
['set', 'short-ban', 'addaction', 'pf']
['multi-set', 'short-ban', 'action', 'pf', [['actionstop', 'pfctl -sr 2>/dev/null | grep -v <tablename> | pfctl -f-\npfctl -t <tablename> -T flush\npfctl -t <tablename> -T kill'], ['actioncheck', ''], ['actionban', '/sbin/pfctl -t <tablename> -T add <ip>'], ['actionstart', 'pfctl -d\npfctl -f /etc/pf.conf\npfctl -e'], ['actionunban', '/sbin/pfctl -t <tablename> -T delete <ip>'], ['protocol', 'tcp'], ['known/multiport', 'any port <port>'], ['known/actiontype', '<multiport>'], ['known/allports', 'any'], ['known/protocol', 'tcp'], ['multiport', 'any port <port>'], ['tablename', 'short'], ['known/tablename', 'short'], ['actiontype', '<multiport>'], ['allports', 'any']]]
['add', 'long-ban', 'auto']
['set', 'long-ban', 'usedns', 'warn']
['set', 'long-ban', 'addlogpath', '/var/log/fail2ban.log', 'head']
['set', 'long-ban', 'maxretry', 3]
['set', 'long-ban', 'addignoreip', '127.0.0.1']
['set', 'long-ban', 'addignoreip', '192.168.100.0/24']
['set', 'long-ban', 'addignoreip', 'XXX.XXX.XXX.XXX']
['set', 'long-ban', 'logencoding', 'utf-8']
['set', 'long-ban', 'bantime', '259200']
['set', 'long-ban', 'ignorecommand', '']
['set', 'long-ban', 'findtime', '604800']
['set', 'long-ban', 'addfailregex', 'NOTICE\\s+\\[short-ban\\]\\s+Ban\\s+<HOST>']
['set', 'long-ban', 'addignoreregex', 'NOTICE\\s+\\[long-ban\\]\\s+Ban\\s+<HOST>']
['set', 'long-ban', 'addaction', 'longban']
['multi-set', 'long-ban', 'action', 'longban', [['actionstop', 'pfctl -sr 2>/dev/null | grep -v <tablename> | pfctl -f-\npfctl -t <tablename> -T flush\npfctl -t <tablename> -T kill'], ['actioncheck', ''], ['actionban', '/sbin/pfctl -t <tablename> -T add <ip>'], ['actionstart', ''], ['actionunban', '/sbin/pfctl -t <tablename> -T delete <ip>'], ['protocol', 'tcp'], ['known/multiport', 'any port <port>'], ['known/actiontype', '<multiport>'], ['known/allports', 'any'], ['known/protocol', 'tcp'], ['multiport', 'any port <port>'], ['tablename', 'long'], ['known/tablename', 'long'], ['actiontype', '<multiport>'], ['allports', 'any']]]
['add', 'forever-ban', 'auto']
['set', 'forever-ban', 'usedns', 'warn']
['set', 'forever-ban', 'addlogpath', '/var/log/fail2ban.log', 'head']
['set', 'forever-ban', 'maxretry', 5]
['set', 'forever-ban', 'addignoreip', '127.0.0.1']
['set', 'forever-ban', 'addignoreip', '192.168.100.0/24']
['set', 'forever-ban', 'addignoreip', 'xxx.xx.xxx.xxx']
['set', 'forever-ban', 'logencoding', 'utf-8']
['set', 'forever-ban', 'bantime', '-1']
['set', 'forever-ban', 'ignorecommand', '']
['set', 'forever-ban', 'findtime', '604800']
['set', 'forever-ban', 'addfailregex', 'NOTICE\\s+\\[long-ban\\]\\s+Ban\\s+<HOST>']
['set', 'forever-ban', 'addignoreregex', 'NOTICE\\s+\\[forever\\]\\s+Ban\\s+<HOST>']
['set', 'forever-ban', 'addaction', 'forever']
['multi-set', 'forever-ban', 'action', 'forever', [['actionstop', ''], ['actioncheck', ''], ['actionban', '/sbin/pfctl -t forever -T add <ip>'], ['actionstart', ''], ['actionunban', ''], ['known/tablename', 'forever'], ['tablename', 'forever']]]
['add', 'vnc', 'auto']
['set', 'vnc', 'usedns', 'warn']
['set', 'vnc', 'addlogpath', '/var/log/system.log', 'head']
['set', 'vnc', 'maxretry', 3]
['set', 'vnc', 'addignoreip', '127.0.0.1']
['set', 'vnc', 'addignoreip', '192.168.100.0/24']
['set', 'vnc', 'addignoreip', '212.45.6.242']
['set', 'vnc', 'logencoding', 'utf-8']
['set', 'vnc', 'bantime', '600']
['set', 'vnc', 'ignorecommand', '']
['set', 'vnc', 'findtime', '3600']
['set', 'vnc', 'addfailregex', '.*: Authentication: FAILED :: User Name: .* :: Viewer Address: <HOST> :: Type: .*$']
['set', 'vnc', 'addaction', 'pf']
['multi-set', 'vnc', 'action', 'pf', [['actionstop', 'pfctl -sr 2>/dev/null | grep -v <tablename> | pfctl -f-\npfctl -t <tablename> -T flush\npfctl -t <tablename> -T kill'], ['actioncheck', ''], ['actionban', '/sbin/pfctl -t <tablename> -T add <ip>'], ['actionstart', 'pfctl -d\npfctl -f /etc/pf.conf\npfctl -e'], ['actionunban', '/sbin/pfctl -t <tablename> -T delete <ip>'], ['protocol', 'tcp'], ['known/multiport', 'any port <port>'], ['known/actiontype', '<multiport>'], ['known/allports', 'any'], ['known/protocol', 'tcp'], ['multiport', 'any port <port>'], ['tablename', 'short'], ['known/tablename', 'short'], ['actiontype', '<multiport>'], ['allports', 'any']]]
['start', 'short-ban']
['start', 'long-ban']
['start', 'forever-ban']
['start', 'vnc']
```

#### Any customizations done to /etc/fail2ban/ configuration

only added couple of  jails for incremental ban based on fail2ban.log parsing

#### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

I'm unable to provide DEGUG level log, because fail2ban.filterpoll spams fail2ban.log with 
`DEBUG   /var/log/fail2ban.log has been modified` message constantly (bug?)
```
2016-08-28 13:52:17,615 fail2ban                [251]: CRITICAL Unhandled exception in Fail2Ban:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/fail2ban/server/jailthread.py", line 69, in run_with_except_hook
    run(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filterpoll.py", line 111, in run
    self.getFailures(filename)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 749, in getFailures
    self.processLineAndAdd(line)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 417, in processLineAndAdd
    for element in self.processLine(line, date)[1]:
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 412, in processLine
    tupleLine, date, returnRawHost, checkAllRegex)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 481, in findFailure
    dateTimeMatch = self.dateDetector.getTime(timeText, tupleLine[3])
  File "/Library/Python/2.7/site-packages/fail2ban/server/datedetector.py", line 227, in getTime
    date = template.getDate(line, timeMatch[0])
  File "/Library/Python/2.7/site-packages/fail2ban/server/datetemplate.py", line 247, in getDate
    return reGroupDictStrptime(groupdict), dateMatch
  File "/Library/Python/2.7/site-packages/fail2ban/server/strptime.py", line 178, in reGroupDictStrptime
    year, month, day, hour, minute, second, fraction)
ValueError: year is out of range
2016-08-28 13:52:17,641 fail2ban                [251]: CRITICAL Unhandled exception in Fail2Ban:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/fail2ban/server/jailthread.py", line 69, in run_with_except_hook
    run(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filterpoll.py", line 111, in run
    self.getFailures(filename)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 749, in getFailures
    self.processLineAndAdd(line)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 417, in processLineAndAdd
    for element in self.processLine(line, date)[1]:
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 412, in processLine
    tupleLine, date, returnRawHost, checkAllRegex)
  File "/Library/Python/2.7/site-packages/fail2ban/server/filter.py", line 481, in findFailure
    dateTimeMatch = self.dateDetector.getTime(timeText, tupleLine[3])
  File "/Library/Python/2.7/site-packages/fail2ban/server/datedetector.py", line 227, in getTime
    date = template.getDate(line, timeMatch[0])
  File "/Library/Python/2.7/site-packages/fail2ban/server/datetemplate.py", line 247, in getDate
    return reGroupDictStrptime(groupdict), dateMatch
  File "/Library/Python/2.7/site-packages/fail2ban/server/strptime.py", line 178, in reGroupDictStrptime
    year, month, day, hour, minute, second, fraction)
ValueError: year is out of range
```

#### Relevant lines from monitored log files in question:

```
```

**Comment by sebres** [comment]
> I'm unable to provide DEGUG level log, because fail2ban.filterpoll spams fail2ban.log with
DEBUG /var/log/fail2ban.log has been modified message constantly (bug?)

No not a bug, you can not use DEBUG as long as you parse f2b.log, because then your "recidive" jails will always find modifications, because they are write it self.
To test in DEBUG mode, just disable this jails...

To the issue: I suspect it could be some "strange" date-time pattern, where something will be found as year. Can you check the newest fix #1512 included in your installation?
Otherwise try it with DEBUG without recidive jails

**Comment by movalex** [comment]
Hi!
I disabled incremental jails but nothing appeared in log, just the same ValueError on launch. 
Only if I disable all jails, the error will not be shown.
Strangely, it only appears on computer launch. If I restart fail2ban-client a bit later, it launches without error and works well.
I checked #1512 - it seems like I have the same build installed (datetemplate.py file has the same changes)

**Comment by sebres** [comment]
Thx,
well, the fix does the check of word boundaries, but it can be still the wrong recognition of date time, so the large number like 9999 will be recognized as year... 
You can currently only direct specify the date pattern for the jails, that really used in the log format. I'll make two things as new PR:
- log the value and the pattern, to know which it is
- move this pattern in the error case to bottom of template list and repeat auto recognition of date template (any case better as boom)

Can you look in your logs, where it should after start find the large or too small number implicit before date in some format without year?

**Comment by sebres** [comment]
> just the same ValueError on launch. 

Have you tried debug mode? Any messages explicit before boom?

**Comment by yarikoptic** [comment]
it might be worth just running the server under pdb (python debugger) and intospect from the point of failure what is happening.  smth like (didn't try)

in one shell
```shell

pdb `which fail2ban-server` -f --loglevel INFO --logtarget=STDOUT -v
```
and if we are in luck and it stops at the exception -- go up and see on which line it failed etc

**Comment by movalex** [comment]
No messages in DEBUG mode except this one:
```
2016-08-28 20:37:05,368 fail2ban.filterpoll     [240]: DEBUG   /var/log/system.log has been modified
2126 2016-08-28 20:37:06,705 fail2ban                [240]: CRITICAL Unhandled exception in Fail2Ban:
2127 Traceback (most recent call last): ...
```
By the way. I think you're right about wrong date. Before upgrade to v0.10 (which in fact suits me much more than 0.9) I saw this error message on launch (found in old log)

`2016-08-16 10:01:15,896 fail2ban.filter         [258]: ERROR   findFailure failed to parse timeText: 0000-12-30 00:0       0:00`
But in v0.9 this error never crashed fail2ban.

**Comment by sebres** [comment]
Thx, that helps a bit... I'll try to find and fix it.

**Comment by movalex** [comment]
Hi! I've found that piece in system.log that caused this error:
```
 Aug 28 20:37:04 server Keychain Circle Notification[402]: writeToStorage plist={
             absentCircleWithNoReason = 0;
             applicationDate = "0000-12-30 00:00:00 +0000";
             lastCircleStatus = "-1";
             lastWritten = "2016-08-28 17:37:04 +0000";
             pendingApplicationReminder = "4001-01-01 00:00:00 +0000";
             pendingApplicationReminderInterval = 172800;
}
```
This "Keychain Circle Notification" is some strange piece of opensource software in OSX, which is responsible for handling iCloud Keychain Approval. And I'm not sure if I can safely disable it or delete :(  iCloud is not even enabled on my server.

**Comment by sebres** [comment]
really so with newlines? I though, the newlines will be escaped as `\n` in message if someone writes it in syslog... weird.

**Comment by movalex** [comment]
I copied this from vim. 
That's how it looks like: 
![f2ban_error](https://cloud.githubusercontent.com/assets/11698866/18081995/8938c9e6-6ea5-11e6-965b-c882ef7ad4d6.JPG)

**Comment by sebres** [comment]
Very very evil... (I mean the newlines)

**Comment by sebres** [comment]
So it should be fixed with #1539, but...

BTW. I'll recommend you not to parse system.log (better to switch on the rsyslog-filtering resp. facility and parse corresponding log only). I don't know whether OSX uses rsyslog (or something similar), but it is very easy on linux systems (and default on). For example, normally sshd should additionally log into `auth.log` also (and I think, it will not contain such scurrilous log-messages, that you've in your system.log)...
Additionally, it is very evil to write unescaped new-lines (mean foreign user-input) in log-files, that will be monitored, because it opens some "undesirable" possibilities for theoretical attackers, you know what I mean ;))
Additionally parsing of facility-targeted log-files should be faster and fewer system load intensive (because fewer parasitic messages occurrence in sense of fail2ban)
